install.py

```diff
@@ -0,0 +1,235 @@
+#!/usr/bin/env python3
+from dataclasses import dataclass
+from enum import Enum
+from typing import List, Optional
+
+import argparse
+import os
+import subprocess
+import sys
+import yaml
+import shutil
+
+from build_image import InstallTarget
+
+
+class DependencyMechanism(Enum):
+    NONE = "none"
+    SHOW = "show"
+    APT = "apt"
+    BREW = "brew"
+
+
+@dataclass
+class OfrakInstallConfig:
+    packages_paths: List[str]
+    python_command: str
+    install_target: InstallTarget
+    dependency_mechanism: DependencyMechanism
+    dep_install: List[str]
+    quiet: bool
+
+
+def main():
+    config = parse_args()
+
+    for package_path in config.packages_paths:
+        check_package_contents(package_path)
+
+    if not config.quiet:
+        print(f"*** Checking whether npm and rollup are installed")
+    check_executable(config, "npm")
+    check_executable(config, "rollup")
```

Comment:
> Did you consider building a Makefile target to do this? What I'm envisioning is something like:
> 
> ```
> OFRAK_CONFIG=ofrak-dev.yml PYTHON_PATH=python3 make develop
> ```
@whyitfor , I can add `OFRAK_CONFIG`, but then would probably want to do the same for the `build_image` targets? Right now we have a few hardcoded targets with specific configs, rather than `OFRAK_CONFIG`, so I kept the same for consistency.

> 1. Parse the yaml to get list of packages to install
> 2. Call `make develop/install` in the respective directory

At least the above steps would presumably still require a python script?

> 3. There could be an optional argument at the end to run apt/brew commands

Is there any benefit to handling the apt vs brew vs just print the suggestion logic in Makefile rather than in Python?

> I also wonder if we need an automated script to install from source -- this is going to increase the amount of things taht need to be tested and maintained.

Reply:
@rbs-jacob changed to use `npm install` in place of explicitly making sure rollup is installed